pkg/overlay/cache.go

```diff
@@ -267,6 +256,33 @@ func (cache *Cache) GetAll(ctx context.Context, ids storj.NodeIDList) (_ []*Node
 	return cache.db.GetAll(ctx, ids)
 }
 
+// UnreliableOrOffline filters a set of nodes to unhealth or offlines node, independent of new
+func (cache *Cache) UnreliableOrOffline(ctx context.Context, nodeIds storj.NodeIDList) (badNodes map[storj.NodeID]bool, err error) {
```

Comment:
Also return a list rather than a map. Checking against a short list might be faster on average rather than a map and involve less allocations.

Reply:
Fixed